app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/commontask/CommonTaskViewModel.kt

```diff
@@ -240,6 +240,21 @@ class CommonTaskViewModel(appComponent: AppComponent = TaigaApp.appComponent) :
 
     fun addAssignee(user: User) = changeAssignees(user, remove = false)
     fun removeAssignee(user: User) = changeAssignees(user, remove = true)
+    fun assigneeOrWatchToMe(isWatcher: Boolean = false) = viewModelScope.launch {
+        val user = usersRepository.getUser(session.currentUserId.value)
```

Comment:
Unnecessary call to API, since we need to know only user id to perform this action. Better to use direct API (assign / watch) call with user id from session

Reply:
But I need `User` class for `addAssignee` and `addWatcher` function, add without this calling I don't have an opportunity to get `User` class, doesn't it?

Reply:
Well, I think you might modify `changeWatchers` and `changeAssignees` to accept user id instead of `User` object (since these methotds actually need only user id)